test/data/pipeline-template-with-pipeline-setting.json

```diff
@@ -0,0 +1,39 @@
+{
+    "id":111,
+    "version": "1.0.0",
+    "templateId": 1234,
+    "description": "Template for building a NodeJS module\nInstalls dependencies and runs tests\n",
+    "config": {
```

Comment:
Please add tests to cover below scenarios

1. The fields exists in the pipeline using the template but not in the template
2. The fields does not exists in the pipeline using the template but exists in the template
3. The fields exists both in the pipeline using the template and in the template. We need to cover sub-scenarios like adding new entry vs replacing an entry vs appending to an entry, etc

Looks like the current test just covers `1`.